practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/repository/HapiFhirRepository.java

```diff
@@ -33,28 +33,32 @@
      * @param practitioner the practitioner to save.
      * @return id of the saved practitioner
      */
-    public String savePractitioner(Practitioner practitioner) {
+    public String createPractitioner(final Practitioner practitioner) {
+
         // Log the request
         if (logger.isInfoEnabled()) {
             logger.info(FhirRepo.SAVE_REQUEST.message(), fhirContextWrapper.prettyPrint(practitioner));
         }
 
-        Bundle responseBundle;
+        final Bundle responseBundle;
         try {
             responseBundle = fhirContextWrapper.executeTransaction(bundle(practitioner));
         } catch (FhirServerResponseException e) {
             final String message = FhirRepo.FAILED_TO_SAVE_PRACTITIONER.message();
             throw new RestException(message, e);
         }
-        IBaseResource responseElement = extractResponseResource(responseBundle);
+        IBaseResource responseElement = fhirContextWrapper.toSingleResource(responseBundle);
 
         // Log the response
         if (logger.isInfoEnabled()) {
             logger.info(FhirRepo.SAVE_RESPONSE.message(), fhirContextWrapper.prettyPrint(responseElement));
         }
 
-        //TODO: replace with the actual id return
-        return practitioner.getIdElement().getValue();
+        return getUnqualifiedResourceId(responseElement);
+    }
+
+    private String getUnqualifiedResourceId(final IBaseResource responseElement) {
+        return responseElement.getIdElement().toUnqualified().getIdPart();
```

Comment:
1. I haven't used toUnqualified since getIdPart looked like it does exactly what I needed... Are you sure we need that step? Also, perhaps the method name should reflect the main idea of getting the IdPart?
2. Please update savePractitionerRole with this too.